packages/typespec-client-generator-core/src/http.ts

```diff
@@ -716,6 +716,10 @@ function findMapping(
     ) {
       return methodParam;
     }
+    // If the service parameter is a body parameter, try to see if we could find a method parameter with same type of the body parameter.
+    if (serviceParam.kind === "body" && serviceParam.type === methodParam.type) {
+      return methodParam;
+    }
```

Comment:
I am not sure why the BFS approach produce different result than this "shortcut" approach.

Reply:
The BFS only compare the parameter/property, but for this special case, we use the parameter's type directly.